### DISCUSSION TITLE
Como está o cenário de nodeJS no backend?!

### DISCUSSION THREAD

**Discussion by Cleysonlb** [frontendbr/forum#1059] [Discussão]
Fala galera beleza?! Tava batendo um papo sobre node com um brother meu e surgiu algumas duvidas de como esta sendo usado o node hoje no backend no mercado.
- Tem algum framework predominante?
- MVC ou outro?
- JS funcional no backend?

**Comment by mahenrique94** [comment]
Acredito que um dos mais famosos seja Express e Restfy, para trabalhar com API’s seguindo o padrão REST.

Para banco de dados tem o Mongoose e Sequelize para ORM se não me engano.

E diversas outras ferramentas, frameworks e bibliotecas para se construir um back-end completo.

Abraços

**Comment by torto** [comment]
Hoje tudo que vc faz com outras linguagens consegue se fazer com node :D

**Comment by Cleysonlb** [comment]
Opa valeu @mahenrique94, ja conheço o express e o sequelize, vou dar uma olhada no Restfy!

**Comment by Murillofilho86** [comment]
Muito bom tipo... Problema que sempre vejo empresas oferecendo vagas pra frontend e pedindo nodejs, acabo ficando confuso... Rs

**Comment by bernardodiasc** [comment]
NodeJS no frontend é lindo demais :D

**Comment by Cleysonlb** [comment]
NodeJS no front hoje é quase obrigatório, com o tanto de bundles tools e ferramentas que rodam em cima do Node...
Minha duvida em si é como o mercado/comunidade estão usando o NodeJS no backend hoje

**Comment by torto** [comment]
Olha trabalhei por dois anos em uma empresa chamada Sprinklr, ela é especialista em monitoramento de redes sociais. Trabalhávamos com muito dados e uma escala bem grande, usávamos em todos nossos projetos nodejs e ele nunca foi um problema.
Então posso dizer que sim, estão usando NodeJS em produção!

**Comment by torto** [comment]
Segue um link com os 10 maiores frameworks em 2018: [clique aqui](https://excellentwebworld.com/most-popular-node-js-frameworks/)

**Comment by Cleysonlb** [comment]
@torto bem legal ver que o SailsJs ta ganhando espaço, acho bem legal a estrutura deles, parecida com a do Rails que eu ja sou familiarizado!

**Comment by mksanti** [comment]
Para quem curte algo mais parrudo e do estilo Laravel, tem esse cara aqui que é mt promissor.
[https://adonisjs.com/](https://adonisjs.com/)

**Comment by fakenickels** [comment]
Koa+GraphQL+Node+Postgres ou Mongo ftw

**Comment by ninetails** [comment]
Relacionado, tava vendo esses dias sobre algo pra servidor que fosse mais "funcional", tentando evitar callback hell e/ou "middleware hell" <sup><sub>(não que realmente exista um termo popularizado sobre isso, foi apenas uma crítica)</sub></sup>, curti bastante usar o [zeit/micro](https://github.com/zeit/micro).

**Comment by felipefialho** [comment]
@luizbills Curiosidade, vi que negativou o @torto falando que é possivel fazer tudo no Node hoje, vê muitos problemas ou coisas que ainda não são possíveis? (ps, fiquei curioso mesmo)

**Comment by lucianobritodev** [comment]
Cara, node é top! Já usei muito com NestJS e TypeORM (para BD Relacional) ou Mongoose (para MongoDB), mas vejo muito a galera ainda raiz usando o Express com Sequelize.
